src/repositories/user.repository.ts

```diff
@@ -0,0 +1,52 @@
+import { v4 as uuidv4 } from "uuid";
+
+import { User } from "../models";
+
+export class UserRepository {
+  private users = [] as User[];
+
+  getUsers(): User[] {
+    return this.users.filter((user) => !user.isDeleted);
+  }
+
+  getUserByID(userId: string): User {
+    const index = this.findUserIndex(userId);
+
+    return this.users[index];
+  }
+
+  addUser(user: User): User {
```

Comment:
Please split checking user and addicting functionality (Single responsiblility)

Reply:
One more method is added. Thank you.